core/src/main/scala/org/ensime/core/AnalyzerManager.scala

```diff
@@ -0,0 +1,385 @@
+package org.ensime.core
+
+import java.nio.file.Path
+
+import akka.actor.{ Actor, ActorLogging, ActorRef, Props, Stash }
+import akka.event.LoggingReceive.withLabel
+import org.ensime.api._
+import org.ensime.config._
+import org.ensime.indexer.SearchService
+import org.ensime.util.FileUtils.{ convertToPath, toSourceFileInfo }
+import org.ensime.util.ensimefile._
+import org.ensime.util.file.File
+import org.ensime.vfs.`package`.EnsimeVFS
+
+import scala.collection.{ breakOut, mutable }
+
+trait ModuleFinder {
+  implicit val config: EnsimeConfig
+
+  private val module_srcs = for {
+    project <- config.projects
+    source <- project.sources
+  } yield (project, source.toPath)
+
+  def getModule(path: Path): Option[EnsimeProject] = {
+    val module_tuple = module_srcs.find {
+      case (name, source) => path.toString.startsWith(source.toString)
+    }
+    module_tuple match {
+      case Some((module, _)) =>
+        Some(module)
+      case None =>
+        None
+    }
+  }
+
+  def getModule(file: RichEnsimeFile): Option[EnsimeProject] = getModule(file.toPath)
+  def getModule(fileInfo: SourceFileInfo): Option[EnsimeProject] = getModule(richEnsimeFile(fileInfo.file))
+  def getModule(file: File): Option[EnsimeProject] = getModule(richEnsimeFile(RawFile(file.toPath)))
+  def getModule(file: Either[File, SourceFileInfo]): Option[EnsimeProject] = file match {
+    case Left(f) => getModule(f)
+    case Right(sourceFileInfo) => getModule(sourceFileInfo)
+  }
+}
+
+class LoadedFilesData(module: EnsimeProject) {
+  /**
+   * typeCheckedAll  flag is true when a TypecheckModule or TypecheckAllReq was more recent than an UnloadAllReq or UnloadModuleReq
+   * unloadedAll     flag is true when a UnloadAllReq or UnloadModuleReq was more recent than an TypecheckModule or TypecheckAllReq
+   * If we haven't seen any, both flag are false
+   * files           means different when different flags are set. Either contains the list od files to be loaded or the files to be unloaded
+   * tempFiles       used when both flags are false; gives list of files that were loaded
+   */
+  private var typeCheckedAll = false
```

Comment:
I'm not keen on non-actor code storing state like this. Can't you refactor this is as an immutable `case class` that lives in an actor?